### ISSUE TITLE
[MU4 Issue] Muse Hub won't open in Linux Mint

### ISSUE THREAD

**Issue by rgreen5** [musescore/MuseScore#13909]
**Describe the bug**

After installing Muse Hub from the Deb file, nothing happens after clicking on the menu item "Muse Hub".

**Platform information**

Linux Mint 20.1 (Cinnamon).

**Comment by llui85** [comment]
I experienced this briefly when installing on Debian as well. My theory is the background service isn't starting immediately. Do you get any output if you run `/usr/bin/muse-hub` in a terminal? Maybe try `sudo /usr/bin/muse-hub-helper` as well if that doesn't result in anything.

**Comment by MarcSabatella** [comment]
I also found it necessary to start the service manually - "sudo muse-hub-service" ("service", not "helper").

**Comment by rgreen5** [comment]
Tried Marc's suggestion, "sudo muse-hub-service", in the terminal.

Got the following output:

2022-10-23 15:05:43.6004|DEBUG|Microsoft.Extensions.Hosting.Internal.Host|Hosting starting
2022-10-23 15:05:43.7368|INFO|Muse.Service.Interop.ServiceCore|Setting SSL_CERT_DIR to /usr/lib/ssl/certs
2022-10-23 15:05:43.7769|INFO|Muse.Service.Interop.ServiceCore|Setting SSL_CERT_FILE to /usr/lib/ssl/certs/ca-certificates.crt
2022-10-23 15:05:43.7769|INFO|Muse.Service.Interop.ServiceCore|Calling init
2022-10-23 15:05:43.7803|ERROR|Muse.Service.Interop.ServiceCore|Could not read from backing file! May not exist?

2022-10-23 15:05:43.7803|INFO|Muse.Service.Interop.ServiceCore|Initial state callback for: Muse Hub
2022-10-23 15:05:43.7803|DEBUG|Muse.Service.Interop.ServiceCore|[MuseDownloaderInitialStateCallbackHandler] ID: hub, Name: Muse Hub, Item Type: Application, State: Installed, Version: 0.9.7.273
2022-10-23 15:05:43.7850|DEBUG|Muse.Common.Services.MuseHubStateManager|Updating Item State: Type: Application, Id: hub, Name: Muse Hub, Version: 0.9.7.273, Progress: 0, State: Installed, BPS: 0, Seconds Remaining: 0
2022-10-23 15:05:43.8222|ERROR|Muse.Service.Interop.ServiceCore|Couldn't load resume data for Muse Hub from hub.resume
unexpected end of file in bencoded string [bdecode:3]
2022-10-23 15:05:43.8250|INFO|Muse.Service.Services.MuseBackgroundService|Muse Background Service Started!
2022-10-23 15:05:43.8250|INFO|Microsoft.Hosting.Lifetime|Application started. Press Ctrl+C to shut down.
2022-10-23 15:05:43.8250|INFO|Microsoft.Hosting.Lifetime|Hosting environment: Production
2022-10-23 15:05:43.8250|INFO|Microsoft.Hosting.Lifetime|Content root path: /home/user1
2022-10-23 15:05:43.8250|DEBUG|Microsoft.Extensions.Hosting.Internal.Host|Hosting started
2022-10-23 15:05:43.8452|INFO|Muse.Service.Services.MuseBackgroundService|Waiting for client to connect...
2022-10-23 15:08:19.8244|ERROR|Muse.Service.Interop.ServiceCore|Other alert: 50: could not map port using UPnP[192.168.0.16]: no router found
2022-10-23 15:08:19.8244|ERROR|Muse.Service.Interop.ServiceCore|Other alert: 50: could not map port using UPnP[192.168.0.16]: no router found

**Comment by sammik** [comment]
Had same problem.
After installing and clicking launcher, nothing happened.
I started it in ternminal by typing `muse-hub`, which works.
After some time (and multiple starts from terminal), it started to respond also to launcher and now opens normaly.

**Comment by rgreen5** [comment]
Managed to successively connect to Musescore hub by opening the "Bin" directory in the terminal then typing "muse-hub".

… and now the "Muse Hub" menu option is working as well.

**Comment by yuki0035** [comment]
> Managed to successively run Muse Hub by opening the "Bin" directory in the terminal then typing "muse-hub".
> 
> … and now the "Muse Hub" menu option is working as well.

Same problem on my ubuntu and I did the same thing, typing muse-hub in the terminal but it doesn't work.

**Comment by MarcSabatella** [comment]
What specifically goes wrong?  Did you also try starting the background service manually, with "sudo muse-hub-service" as mentioned above?

**Comment by yuki0035** [comment]
Yes, but I got an output the same as rgreen5 mentioned above: '' no router found ''.

**Comment by jivy59** [comment]
Mint 20.3 does not use the systemD daemon by default. Musehub seems to require systemD.  You can add systemD to Mint 20.3 with these instructions: https://www.linuxhelp.com/how-to-configure-and-setup-systemd-service-on-linux-mint-20-3

**Comment by Tantacrul** [comment]
Initial support is for Ubuntu 20.04 and later. There will be dedicated support for Linux Mint in a later release.

**Comment by iamtesch** [comment]
The `no router found` error isn't anything to worry about; as long as the service is started and then muse hub starts, you should be good.  We'll clean up the error messages here :)

**Comment by sammik** [comment]
> Initial support is for Ubuntu 20.04 and later. There will be dedicated support for Linux Mint in a later release.

@tantacrul I had same problem in Ubuntu 22.04 - Installed Muse Hub, but after install, when clicked icon, nothing happened.

**Comment by jivy59** [comment]
I did not have enough space on Mint, but I can say with assurance that
MuseHub works well on AVL-MX21. Everyonce in a while I see the message
about the helper service not running. In that case, close the hub, run
"sudo muse-hub.service" in the terminal, then re-open the hub.
Jim Ivy

**Comment by DominicMaas** [comment]
Trying to make this experience nicer, does Linux Mint use the debian package manager? (I see it's based on Ubuntu so I assume so).

What does Mint use instead of systemD?

**Comment by jivy59** [comment]
> Actually Mint 21 does use a "strategically pruned" systemD. And the current Muse-Hub beta runs fine with or without systemD. I believe that the problem that I experienced attempting to install Muse-Hub in Mint was because I was running out of disk-space. When I attempted to place the Muse-Hub files on a separate drive, MS 4.0 beta would not find the location. That has since been fixed and the current development versions do allow the files to be found on a separate partition or USB drive. 

The Linux version of 4.0 enables the Muse-Sounds but it does not enable VST and Effects. Hopefully those features will be present in the Linux version when it is released.

**Comment by llui85** [comment]
Unfortunately not - Linux won't get VST support until 4.1:

https://github.com/musescore/MuseScore/issues/11689#issuecomment-1131925968

**Comment by oktophonie** [comment]
Muse Hub issues are now being handled elsewhere, so will be closed here.
If this issue is still valid, please raise it [here](https://musehub.zendesk.com/).
You can link to this GitHub issue if it provides useful context or discussion.
